Plain reimplementation of the chain convex hull algorithm in pure C.

fpl opened the pull request:
This is a strict reimplementation of the hull algorithm in C. I also have a 'cversion' branch in my repository that includes both C and C++ with a slight difference in the names (_ pre-pended to C ones), a pair of test programs to check for performances and simple test cases.

fpl committed:
Plain reimplementation of the chain convex hull algorithm in pure C.

fpl commented:
Additionally, I did not check if a static definition for some internal functions would be appropriate (orientation helper for instance).

kempenep commented:
Hi Francesco,

Thanks for your efforts. I accepted the pull requests and added a
correction for myhull.c (missing equal sign in declaration of
checkLocationNr). I also updated the version, now set to 1.1.7.

Any news on the integration of jiplib and pyjeo in Debian?

All the best,

Pieter.

Op do 6 mrt 2025 om 13:19 schreef Francesco Paolo Lovergine <
***@***.***>:

> Additionally, I did not check if a static definition for some internal
> functions would be appropriate (orientation helper for instance).
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ec-jrc/jeolib-miallib/pull/5#issuecomment-2703689341>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC3INSLF4Q2KACCIUFVNFAT2TA4LVAVCNFSM6AAAAABYORHMXCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOMBTGY4DSMZUGE>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
> [image: fpl]*fpl* left a comment (ec-jrc/jeolib-miallib#5)
> <https://github.com/ec-jrc/jeolib-miallib/pull/5#issuecomment-2703689341>
>
> Additionally, I did not check if a static definition for some internal
> functions would be appropriate (orientation helper for instance).
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ec-jrc/jeolib-miallib/pull/5#issuecomment-2703689341>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC3INSLF4Q2KACCIUFVNFAT2TA4LVAVCNFSM6AAAAABYORHMXCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOMBTGY4DSMZUGE>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

fpl commented:
On Fri, Mar 07, 2025 at 01:27:25AM -0800, kempenep wrote:
>kempenep left a comment (ec-jrc/jeolib-miallib#5)
>
>Hi Francesco,
>
>Thanks for your efforts. I accepted the pull requests and added a
>correction for myhull.c (missing equal sign in declaration of
>checkLocationNr). I also updated the version, now set to 1.1.7.
>

Yep, a typo at edit time.

>Any news on the integration of jiplib and pyjeo in Debian?
>

For jiplib 1.1.5 I still have problema at test time. I guess there's some issue at Swig level.
All test after the first two always fail.


***@***.***:~/.../debian-gis/pyjeo/jeolib-jiplib/build
$ PYTHONPATH=`pwd`/build/lib: LD_LIBRARY_PATH=`pwd`/build/lib/jiplib:  ctest -V --stop-on-failure 
UpdateCTestConfiguration  from :/home/frankie/debian/debian-gis/pyjeo/jeolib-jiplib/build/DartConfiguration.tcl
Parse Config file:/home/frankie/debian/debian-gis/pyjeo/jeolib-jiplib/build/DartConfiguration.tcl
UpdateCTestConfiguration  from :/home/frankie/debian/debian-gis/pyjeo/jeolib-jiplib/build/DartConfiguration.tcl
Parse Config file:/home/frankie/debian/debian-gis/pyjeo/jeolib-jiplib/build/DartConfiguration.tcl
Test project /home/frankie/debian/debian-gis/pyjeo/jeolib-jiplib/build
Constructing a list of tests
Done constructing a list of tests
Updating test list for fixtures
Added 0 tests to meet fixture requirements
Checking test dependency graph...
Checking test dependency graph end
test 1
       Start  1: test0

1: Test command: /usr/bin/echo "Success"
1: Working Directory: /home/frankie/debian/debian-gis/pyjeo/jeolib-jiplib/build/test
1: Test timeout computed to be: 1500
1: Success
  1/55 Test  #1: test0 ............................   Passed    0.00 sec
test 2
       Start  2: test01

2: Test command: /usr/bin/echo "/usr/bin/python3"
2: Working Directory: /home/frankie/debian/debian-gis/pyjeo/jeolib-jiplib/build/test
2: Test timeout computed to be: 1500
2: /usr/bin/python3
  2/55 Test  #2: test01 ...........................   Passed    0.00 sec
test 3
       Start  3: test1

3: Test command: /usr/bin/python3 "/home/frankie/debian/debian-gis/pyjeo/jeolib-jiplib/test/pytest_open.py" "--input" "/home/frankie/debian/debian-gis/pyjeo/jeolib-jiplib/test/data/modis_nd
vi_2010.tif"
3: Working Directory: /home/frankie/debian/debian-gis/pyjeo/jeolib-jiplib/build/test
3: Test timeout computed to be: 1500
3: Traceback (most recent call last):
3:   File "/home/frankie/debian/debian-gis/pyjeo/jeolib-jiplib/test/pytest_open.py", line 26, in <module>
3:     jim0=jl.createJim(args.input)
3:   File "/home/frankie/debian/debian-gis/pyjeo/jeolib-jiplib/build/build/lib/jiplib/jiplib.py", line 401, in createJim
3:     return Jim_createImg(appDict)
3:            ^^^^^^^^^^^^^
3: NameError: name 'Jim_createImg' is not defined
  3/55 Test  #3: test1 ............................***Failed  Required regular expression not found. Regex=[Success
]  0.20 sec

67% tests passed, 1 tests failed out of 3

Total Test time (real) =   0.21 sec

The following tests FAILED:
           3 - test1 (Failed)
Errors while running CTest
Output from these tests are in: /home/frankie/debian/debian-gis/pyjeo/jeolib-jiplib/build/Testing/Temporary/LastTest.log
Use "--rerun-failed --output-on-failure" to re-run the failed cases verbosely.


-- 
Francesco P. Lovergine | https://lovergine.com/

kempenep commented:
@fpl could you please open a new issue in [jiblib](https://github.com/ec-jrc/jeolib-jiplib) where we can continue the discussion on the build of jiplib? Thanks.
